cms/private_media/tests/test_images.py

```diff
@@ -0,0 +1,369 @@
+        self.private_images = []
+        self.public_images = []
+        for _ in range(3):
+            private_image = ImageFactory(collection=self.root_collection)
+            public_image = ImageFactory(_privacy=Privacy.PUBLIC, collection=self.root_collection)
+            private_image.create_renditions(Filter("fill-10x10"), Filter("fill-20x20"))
+            public_image.create_renditions(Filter("fill-10x10"), Filter("fill-20x20"))
+            self.private_images.append(private_image)
+            self.public_images.append(public_image)
+        PURGED_URLS.clear()
+
+    def test_bulk_make_public(self):
+        """Test the behaviour of PrivateImageManager.bulk_make_public()."""
+        # Three image are already public, so only three should be updated
+        qs = self.model.objects.all().prefetch_related("renditions")
+        with self.assertNumQueries(3):
+            # Query summary:
+            # 1. to fetch the images
+            # 2. to prefetch renditions
+            # 3. to save updates
+            self.assertEqual(self.model.objects.bulk_make_public(qs), 3)
+
+        # No urls should have been purged as part of the update
+        self.assertEqual(PURGED_URLS, [])
+
+        # Verify all images are now public
+        for obj in self.model.objects.only("_privacy", "file_permissions_last_set", "privacy_last_changed"):
+            self.assertIs(obj.privacy, Privacy.PUBLIC)
+            self.assertFalse(obj.file_permissions_are_outdated())
+
+        # Another attempt should result in no updates
+        with self.assertNumQueries(1):
+            # Query summary:
+            # 1. to fetch the images
+            self.assertEqual(self.model.objects.bulk_make_public(self.model.objects.all()), 0)
+
+    def test_bulk_make_private(self):
+        """Test the behaviour of PrivateImageManager.bulk_make_private()."""
+        # Three images are already private, so only three should be updated
+        qs = self.model.objects.all().prefetch_related("renditions")
+        with self.assertNumQueries(3):
+            # Query summary:
+            # 1. to fetch the images
+            # 2. to prefetch renditions
+            # 3. to save updates
+            self.assertEqual(self.model.objects.bulk_make_private(qs), 3)
+
+        # No urls should have been purged as part of the update
+        self.assertEqual(PURGED_URLS, [])
+
+        # Verify all images are now private
+        for image in self.model.objects.only("_privacy", "file_permissions_last_set", "privacy_last_changed"):
+            self.assertIs(image.privacy, Privacy.PRIVATE)
+            self.assertFalse(image.file_permissions_are_outdated())
+
+        # Another attempt should result in no updates
+        with self.assertNumQueries(1):
+            # Query summary:
+            # 1. to fetch the images
+            self.assertEqual(self.model.objects.bulk_make_private(self.model.objects.all()), 0)
+
+
+class TestImageServeView(TestCase):
```

Comment:
Issue: Can you add some tests for `IS_EXTERNAL_ENV` here.

Reply:
Added in [f67ac65](https://github.com/ONSdigital/dis-wagtail/pull/46/commits/f67ac65c035666c2a343dfbb5222dfc577d296ad)